src/Microsoft.ML.Core/Data/DateTime.cs

```diff
@@ -32,22 +32,19 @@ public DvDateTime(SysDateTime sdt)
         /// <summary>
         /// This ctor accepts any value for ticks, but produces an NA if ticks is out of the legal range.
         /// </summary>
-        public DvDateTime(DvInt8 ticks)
+        public DvDateTime(long ticks)
         {
-            if ((ulong)ticks.RawValue > MaxTicks)
-                _ticks = DvInt8.NA;
```

Comment:
If you're still treating this type as if it has an NA then you'll have to do the `long.MinValue` mapping here if it's above range. This will also correct the problem with the assert that I just commented on.